rnd/autogpt_builder/src/components/CustomNode.tsx

```diff
@@ -168,11 +178,13 @@ const CustomNode: FC<NodeProps<CustomNodeData>> = ({ data, id }) => {
       return <div className="connected-input">Connected</div>;
     }
 
-    const renderClickableInput = (displayValue: string) => (
-      <div className="clickable-input" onClick={() => handleInputClick(fullKey)}>
-        {displayValue}
-      </div>
-    );
+  const renderClickableInput = (displayValue: unknown): JSX.Element => (
+    <div className="clickable-input" onClick={() => handleInputClick(fullKey)}>
+      {typeof displayValue === 'string' || typeof displayValue === 'number' || typeof displayValue === 'boolean'
+        ? String(displayValue)
```

Comment:
for the three selected types, I think `JSON.stringify` does the same as `String(v)`/`v.toString()`

so then this whole thing can just be `JSON.stringify(displayValue)`